Sources/CacheAPI-xcache.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines http://www.simplemachines.org
+ * @copyright 2016 Simple Machines and individual contributors
+ * @license http://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 2.1 Beta 3
+ */
+
+if (!defined('SMF'))
+	die('Hacking attempt...');
+
+/**
+ * Our Cache API class
+ * @package cacheAPI
+ */
+class xcache_cache extends cache_api
+{
+	/**
+	 * Checks whether we can use the cache method performed by this API.
+	 *
+	 * @access public
+	 * @param boolean $test Test if this is supported or enabled.
+	 * @return boolean Whether or not the cache is supported
+	 */
+	public function isSupported($test = false)
+	{
+		global $cache_memcached;
+
+		$supported = function_exists('xcache_get') && function_exists('xcache_set') && ini_get('xcache.var_size') > 0;
+
+		if ($test)
+			return $supported;
+		return parent::isSupported() && $supported;
+	}
+
+	/**
+	 * Connects to the cache method. If this fails, we return false, otherwise we return true.
+	 *
+	 * @access public
+	 * @return boolean Whether or not the cache method was connected to.
+	 */
+	public function connect()
```

Comment:
PHP 7 doesn't appear to have xCache so I can't test this, but implanting the code that may possibly work.